src/libs/Navigation/AppNavigator/ReportScreenIDSetter.ts

```diff
@@ -42,6 +44,13 @@ const getLastAccessedReportID = (
 function ReportScreenIDSetter({route, reports, policies, navigation, isFirstTimeNewExpensifyUser = false, reportMetadata}: ReportScreenIDSetterProps) {
     const {canUseDefaultRooms} = usePermissions();
     useEffect(() => {
+        const reportActionID = route?.params?.reportActionID;
+        const regexValidReportActionID = new RegExp(/^\d*$/);
+        if (reportActionID && !regexValidReportActionID.test(reportActionID)) {
+            Navigation.goBack(ROUTES.HOME);
+            return;
+        }
```

Comment:
@samilabud We should move this logic inside `if (route?.params?.reportID) {`

Reply:
And one more thing, lets not use `Navigation.goBack(ROUTES.HOME)`, use `Navigation.navigate` instead as this is causing to go back in the stack to previously opened report instead of staying on the same report.

Reply:
Fixed thanks!, it seems there is a test failing on the master branch, I'm going to keep testing on main until I see it is fixed and let you know.